apps/bot_manager/lib/game_socket_handler.ex

```diff
@@ -0,0 +1,115 @@
+defmodule BotManager.GameSocketHandler do
+  @moduledoc """
+  BotManager socket handler.
+  It handles the communication with the server.
+  """
+
+  use WebSockex, restart: :transient
+  require Logger
+
+  @message_delay_ms 300
+
+  def start_link(%{"bot_client" => bot_client, "game_id" => game_id}) do
+    ws_url = ws_url(bot_client, game_id)
+
+    WebSockex.start_link(ws_url, __MODULE__, %{
+      client_id: bot_client,
+      game_id: game_id
+    })
+  end
+
+  def handle_connect(_conn, state) do
+    send(self(), :move)
+    send(self(), :attack)
+    {:ok, state}
+  end
+
+  def handle_frame(_frame, state) do
+    {:ok, state}
+  end
+
+  def handle_info(:move, state) do
+    Process.send_after(
+      self(),
+      :move,
+      @message_delay_ms
+    )
+
+    timestamp = DateTime.utc_now() |> DateTime.to_unix(:millisecond)
+    {x, y} = create_random_movement()
+
+    game_action =
+      BotManager.Protobuf.GameAction.encode(%BotManager.Protobuf.GameAction{
+        action_type:
+          {:move,
+           %BotManager.Protobuf.Move{
+             direction: %BotManager.Protobuf.Direction{
+               x: x,
+               y: y
+             }
+           }},
+        timestamp: timestamp
+      })
+
+    WebSockex.cast(self(), {:send, {:binary, game_action}})
+
+    {:ok, state}
+  end
+
+  def handle_info(:attack, state) do
+    timestamp = DateTime.utc_now() |> DateTime.to_unix(:millisecond)
+    {x, y} = create_random_movement()
+
+    game_action =
+      BotManager.Protobuf.GameAction.encode(%BotManager.Protobuf.GameAction{
+        action_type:
+          {:attack,
+           %BotManager.Protobuf.Attack{
+             skill: "1",
+             parameters: %BotManager.Protobuf.AttackParameters{
+               target: %BotManager.Protobuf.Direction{
+                 x: x,
+                 y: y
+               }
+             }
+           }},
+        timestamp: timestamp
+      })
+
+    WebSockex.cast(self(), {:send, {:binary, game_action}})
+
+    Process.send_after(self(), :attack, 300, [])
```

Comment:
I think that they are going to try to move and attack at the same time every time.
Because you have a `300` here and `@message_delay_ms`at line `35`.